docs/admin/bootstrap-tokens.md

```diff
@@ -0,0 +1,167 @@
+  usage-bootstrap-signing: base64(true)
+```
+
+The type of the secret must be `bootstrap.kubernetes.io/token` and the name must
+be `bootstrap-token-<token id>`.  It must also exist in the `kube-system`
+namespace.  `description` is a human readable discription that should not be
+used for machine readable information.  The Token ID and Secret are included in
+the data dictionary.
+
+The `usage-bootstrap-*` members indicate what this secret is intended to be used
+for.  A value must be set to `true` to be enabled.
+
+`usage-bootstrap-authentication` indicates that the token can be used to
+authenticate to the API server.  The authenticator authenticates as
+`system:bootstrap:<Token ID>`.  It is included in the `system:bootstrappers`
+group.  The naming and groups are intentionally limited to discourage users from
+using these tokens past bootstrapping.
+
+`usage-bootstrap-signing` indicates that the token should be used to sign the
+`cluster-info` ConfigMap as described below.
+
+The `expiration` data member lists a time after which the token is no longer
+valid.  This is encoded as an absolute UTC time using RFC3339.  The TokenCleaner
+controller will delete expired tokens.
+
+## Token Management with `kubeadm`
+
+You can use the `kubeadm` tool to manage tokens.  It will automatically grab the
+default admin credentials on a master from a `kubeadm` created cluster
+(`/etc/kubernetes/admin.conf`).  You can specify an alternate kubeconfig file
+for credentials with the `--kubeconfig` to the following commands.
+
+* `kubeadm token list` Lists the tokens along with when they expire and what the
+  approved usages are.
+* `kubeadm token create` Creates a new token.
+    * `--description` Set the description on the new token.
+    * `--ttl duration` Set expiration time of the token as a delta from "now".
+      Default is 0 for no expiration.
+    * `--usages` Set the ways that the token can be used.  The default is
+      `signing,authentication`.
+* `kubeadm token delete <token id>|<token id>.<token secret>` Delete a token.
+  The token can either be identified with just an ID or with the entire token
+  value.
+
+## ConfigMap Signing
+
+In addition to authentication, the tokens can be used to sign a ConfigMap.  This
+is used early in a cluster bootstrap process before the client trusts the API
+server.  The signed ConfigMap can be authenicated by the shared token.
+
+The ConfigMap that is signed is `cluster-info` in the `kube-public` namespace.
+The typical flow is that a client reads this ConfigMap while unauthenticated and
+ignoring TLS errors.  It then validates the payload of the ConfigMap by looking
+at a signature embedded in the ConfigMap.
+
+The ConfigMap may look like this:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-info
+  namespace: kube-public
+data:
+  jws-kubeconfig-07401b: eyJhbGciOiJIUzI1NiIsImtpZCI6IjA3NDAxYiJ9..tYEfbo6zDNo40MQE07aZcQX2m3EB2rO3NuXtxVMYm9U
```

Comment:
Tool late to change the value in the secret -- this is baked in to both the BootstrapSigner and kubeadm.  And this is how to verify a secret (essentially reimplement kubeadm) so this level of detail is necessary.  The more user facing docs (kubeadm focused) don't talk about this at all.